types.go

```diff
@@ -57,6 +57,9 @@ type Scope interface {
 
 	// Capabilities returns a description of metrics reporting capabilities.
 	Capabilities() Capabilities
+
+	// InstrumentedCall returns an InstrumentedCall with the given name
+	InstrumentedCall(name string) InstrumentedCall
```

Comment:
Adding this method to this interface is going to cause a major semver bump for tally which we are committed to not doing again.

The only way to provide this as a non-breaking change is to provide it as a module level constructor.  e.g. `func NewInstrumentedCall(scope Scope, name string) InstrumentedCall { ... }`.  

This is the route we'll need to go down to avoid a breaking(major semver) change.

Reply:
To make this API a little more readable we could create a new subpackage called `instrument` and export `func NewCall(scope tally.Scope, name string) Call { ... }`.  That way any future things we'd like to wrap for instrumenting purposes can live in this subpackage.  e.g. You could imagine a `func NewContextExecution(scope tally.Scope, name string) ContextExecution { ... }` that would implement something like:
```go
// public type
type ContextExecution interface {
  // RecordExec records the execution of a context that is cancelable, returns 
  // false if context is not cancelable and cannot be recorded, true if it can and will be
  RecordExec(ctx context.Context) bool
}

// implementation
type contextExecution struct {
  success, error tally.Counter
  latency tally.Timer
}

func NewContextExecution(scope tally.Scope, name string) ContextExecution {
  return &contextExecution{
    success: scope.Tagged(map[string]string{"result_type":"success"}).Counter(name),
    error: scope.Tagged(map[string]string{"result_type":"error"}).Counter(name),
    latency: scope.Timer(fmt.Sprintf("%s.latency", name)),
  }
}

func (c *contextExecution) RecordExec(ctx context.Context) bool {
  doneCh := ctx.Done()
  if doneCh == nil {
    return false // ctx not cancellable, can't record execution
  }
  start := time.Now()
  go func() {
    <-doneCh
    duration := time.Since(start)
    c.success.Inc(1)
    c.error.Inc(1)
    c.latency.Record(duration)
  }()
  return true
}
```

Reply:
I see. Good point. I wanted to keep in parity with go-common, but that isn't worth the major bump required.